### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by letrylesha**
image

### CODE REVIEW

**Review by alinalaktionova — DISMISSED** [review]

File: src/components/App/ContentWrapper/ProfilePage/ProfileInfo/ProfileInfoInterface.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default interface ProfileInfoInterface {
+  email: string;
+  group: string;
+  name: string;
+  tasks: any;
+  time: any;
```
Comment (alinalaktionova) [inline src/components/App/ContentWrapper/ProfilePage/ProfileInfo/ProfileInfoInterface.ts:6]: Why any????

File: src/store/profile/ProfileReducer.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export const initialState = {
+  profileData: null
+};
+
+export const profileReducer = (state = initialState, action: any) => {
+  switch (action.type) {
+    case profileTypes.SAVE_PROFILE_DATA:
+      console.log('SAVE_from_reducer');
+      console.log(action.payload);
```
Comment (alinalaktionova) [inline src/store/profile/ProfileReducer.ts:11]: remove console.log please

File: src/sagas/profile/profile.sagas.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { takeLatest, call, put } from 'redux-saga/effects';
+import { profileTypes } from '../../store/profile/ProfileType';
+import { db } from '../../components/App/Auth/firebase.config';
+import { saveProfileData } from '../../store/profile/ProfileActions';
+import Cookies from "js-cookie";
+
+function getData(data: any) {
+  const user = data.data()
+  return user;
```
Comment (alinalaktionova) [inline src/sagas/profile/profile.sagas.ts:9]: redundant user variable

File: src/components/App/ContentWrapper/HomePage/Tiles/TileItem/TileItem.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+const TileItem: React.FC<Props> = ({ tile }) => {
+    return (
+      <div className="tiles__tile" data-testid="bg">
+      <h3>{ tile.name }</h3>
+      <i className={ `fas ${tile.icon}` }></i>
+      <p>{ tile.text }</p>
+    </div>
+  )
```
Comment (drews-believer) [inline src/components/App/ContentWrapper/HomePage/Tiles/TileItem/TileItem.tsx:16]: Prettier pls

File: src/components/App/ContentWrapper/ProfilePage/ProfileInfo/ProfileInfoInterface.ts (hunk 1 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/ProfilePage/ProfileInfo/ProfileInfoInterface.ts:6]: oh, sorrry((((

File: src/sagas/profile/profile.sagas.ts (hunk 3 above)
Comment (letrylesha) [reply src/sagas/profile/profile.sagas.ts:9]: ...

**Review by CBond05 — COMMENTED** [review]
Merge into master?

File: src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+interface Props {
+  tiles: TileInterface[];
+}
+
+const Tiles: React.FC<Props> = ({ tiles }) => {
+  return (
+    <div className="tiles-container">
+      <StyledTiles>
+        {
+          tiles.map((tile: any) => <TileItem tile={tile} key={tile.id} />)
```
Comment (drews-believer) [inline src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx:31]: tile: any.

File: src/sagas/auth/auth.sagas.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      ...action.payload
+    });
+
+    if (!isSuccess) {
+      return;
+    }
+
+    yield call(saveDataInState, { ...userAuthData, error: null, isAuth: true });
+  } catch (e) {
+    // error handle in future
```
Comment (CBond05) [inline src/sagas/auth/auth.sagas.tsx:82]: any error handling ?

File: src/components/App/ContentWrapper/HomePage/HomePage.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+const StyledTitleH1 = styled.h1`
+  color: #232e3d;
+  @media (max-width: 1200px) {
+    text-align: center;
+  }
+`;
+
+const HomePage = () => {
+
+  const [tiles, setTiles] = useState<any[]>([]);
```
Comment (drews-believer) [inline src/components/App/ContentWrapper/HomePage/HomePage.tsx:23]: any[]

File: src/components/App/ContentWrapper/ProfilePage/ProfilePage.test.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+import configureStore from "redux-mock-store";
+import "@testing-library/jest-dom/extend-expect";
+
+const mockStore = configureStore([]);
+afterEach(cleanup);
+describe("AuthPopUp", () => {
+  it('Render user info', () => {
+
+    })
+});
```
Comment (drews-believer) [inline src/components/App/ContentWrapper/ProfilePage/ProfilePage.test.tsx:11]: empty test)

**Review by danill1278 — COMMENTED** [review]
Problems were mentioned above)

File: src/sagas/auth/auth.sagas.tsx (hunk 6 above)
Comment (letrylesha) [reply src/sagas/auth/auth.sagas.tsx:82]: We've not made UI for error handling

File: src/components/App/ContentWrapper/HomePage/Tiles/TileItem/TileItem.tsx (hunk 4 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/HomePage/Tiles/TileItem/TileItem.tsx:16]: class Andrey implements Proverylka)))

File: src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx (hunk 5 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx:31]: Fixed

File: src/components/App/ContentWrapper/HomePage/Tiles/TileItem/TileItem.tsx (hunk 4 above)
Comment (drews-believer) [reply src/components/App/ContentWrapper/HomePage/Tiles/TileItem/TileItem.tsx:16]: smeshno chuchut'))

File: src/components/App/ContentWrapper/ProfilePage/ProfilePage.test.tsx (hunk 8 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/ProfilePage/ProfilePage.test.tsx:11]: Fixed bro

File: src/components/App/ContentWrapper/HomePage/HomePage.tsx (hunk 7 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/HomePage/HomePage.tsx:23]: Fixed bro

**Pull request closed without merging**